app/services/intercom_service.rb

```diff
@@ -3,147 +3,130 @@
 class IntercomService
   def self.create_intercom_message_from_portal_message(portal_message, inform_of_handoff: false)
     client = portal_message.client
-    contact_id_from_client_id = contact_id_from_client_id(client.id)
-    contact_id = contact_id_from_client_id.present? ? contact_id_from_client_id : create_or_update_intercom_contact(portal_message).id
-
-    if contact_id_from_client_id.present? && most_recent_conversation(contact_id).present?
-      reply_to_existing_intercom_thread(contact_id, portal_message.body)
-    else
-      create_new_intercom_thread(contact_id, portal_message.body)
-      if inform_of_handoff
-        contact_info = client.intake.contact_info_filtered_by_preferences
-        send_handoff_email(client) if contact_info[:email]
-        send_handoff_sms(client) if contact_info[:sms_phone_number]
-      end
-    end
+    create_intercom_message(
```

Comment:
We now need to allow for forwarding without a message object. Thus, a lot of the supporting methods needed to either be duplicated (because they expected an object to reference, and we only have attributes) or consolidated, or I needed to create some kind of wrapper object for a non-client message object. 

To me, the logic for this whole thing can be simplified by looking at what each of these things has in common: a message body, contact method, and optionally a client, and passing those into the consuming method as params.